nx_cugraph/convert_matrix.py

```diff
@@ -276,3 +284,95 @@ def from_scipy_sparse_array(
     if inplace:
         return create_using._become(G)
     return G
+
+
+@networkx_algorithm(version_added="25.06", fallback=True)
+def to_numpy_array(
+    G,
+    nodelist=None,
+    dtype=None,
+    order=None,
+    multigraph_weight=sum,
+    weight="weight",
+    nonedge=0.0,
+):
+    """MultiGraphs are not yet supported"""
+    # print(" ==> hello!! dispatched to nxcg!")
+    G = _to_graph(G, weight, 1, _get_float_dtype(dtype))
```

Comment:
`_get_float_dtype` probably isn't needed here. If `dtype is None`, maybe use a default of `dtype = np.float64`. The dtype may be int, complex, structured/record, etc., which `_get_float_dtype` probably doesn't work with. Please lmk if I'm wrong and there's a good reason you're using this ;)

Also, `_to_graph` only preserves a single edge attribute. When `dtype` is a structured dtype that indicates many edge attributes, we should use `nxcg.from_networkx` to preserve all required edges.